File: Sorting_algorithms/Kotlin/insertion_sort.kt
fun insertion_sort (arr: IntArray) {
    val arr_lenght = arr.size
    for (i in 1 until arr_lenght) {
        var lowest_value = arr[i]
        var j = i-1
        while (j >= 0 && arr[j] > lowest_value) {
            arr[j+1]=arr[j]
            j--
        }
        arr[j+1]=lowest_value
    }
}

fun main () {
    val arrList = intArrayOf(11, 30, -1, 2, 7, 12, 13, 20, 8)
    println("\nBefore Sort: ${arrList.joinToString(", ")}")
    insertion_sort(arrList)
    print("After Sort: ${arrList.joinToString(", ")}\n")
}

File: Sorting_algorithms/python/src/insertion_sort.py
def insertion_sort(arr:list):
    arr_len = len(arr)
    for i in range( 1, arr_len):
        lowest_value = arr[i]
        j = i-1
        while j >= 0 and arr[j] > lowest_value:
            arr[j+1] = arr[j]
            j-=1
        arr[j+1] = lowest_value
    return arr

if __name__ == "__main__":
    arr = [11, 30, -1, 2, 7, 12, 13, 20, 8]
    print(f"\nBefore Sort: {arr}")
    print(f"After Sort: {insertion_sort(arr=arr)}\n")